daffodil-core/src/main/scala/org/apache/daffodil/dsom/ComplexTypes.scala

```diff
@@ -52,8 +55,26 @@ abstract class ComplexTypeBase(xmlArg: Node, parentArg: SchemaComponent)
     xmlChildren.flatMap {
       xmlChild =>
         {
-          val g = ModelGroupFactory(xmlChild, this, 1, false) // discards unwanted text nodes also.
-          g
+          xmlChild match {
+            case <annotation>{ annotationChildren @ _* }</annotation> => {
+              val dais = annotationChildren.find { ai =>
+                ai.attribute("source") match {
+                  case Some(n) => n.text.contains("ogf") && n.text.contains("dfdl")
+                  case _ => false
+                }
+              }
+              if (dais != None) {
+                this.SDW(WarnID.InvalidAnnotationPoint, "complexType is not a valid annotation point. Annotation dropped.")
```

Comment:
I prefer the term "ignored" to "dropped"

Reply:
Updates made